src/QMCDrivers/QMCCorrelatedSamplingLinearOptimize.cpp

```diff
@@ -211,8 +211,11 @@ bool QMCCorrelatedSamplingLinearOptimize::run()
     }
     if (MinMethod == "rescale")
     {
-      for (int i = 0; i < numParams; i++)
-        bestParameters[i] = optTarget->Params(i) = currentParameters[i] + Lambda * currentParameterDirections[i + 1];
+      for (int i = 0; i < numParams; i++) {
+        //bestParameters[i] = optTarget->Params(i) = currentParameters[i] + Lambda * currentParameterDirections[i + 1];
+        optTarget->Params(i) = currentParameters[i] + Lambda * currentParameterDirections[i + 1];
+        bestParameters[i] = std::real(optTarget->Params(i));
```

Comment:
I believe if you don't put std::real, the code doesn't compile but if you put std::real in.
The code is logically broken here. Because you are forcing a real-only optimizer to work.
So please add a FIXME note in every place like this where you think a fix will be needed.

Reply:
I have added a FIXME note.